sdk/test/trace/probability_sampler_test.cc

```diff
@@ -0,0 +1,224 @@
+#include "opentelemetry/sdk/trace/samplers/probability.h"
+#include "src/common/random.h"
+
+#include <gtest/gtest.h>
+#include <cstdlib>
+#include <ctime>
+
+using opentelemetry::sdk::common::Random;
+using opentelemetry::sdk::trace::Decision;
+using opentelemetry::sdk::trace::ProbabilitySampler;
+using opentelemetry::trace::SpanContext;
+
+namespace
+{
+/*
+ * Helper function for running probability sampler tests.
+ * Given a span context, sampler, and number of iterations this function
+ * will return the number of RECORD_AND_SAMPLE decision based on randomly
+ * generated traces.
+ *
+ * @param context a required valid span context
+ * @param sampler a required valid sampler
+ * @param iterations a requried number specifying the number of times to
+ * generate a random trace_id and check if it should sample using the provided
+ * provider and context
+ */
+int RunShouldSampleCountDecision(SpanContext &context,
+                                 ProbabilitySampler &sampler,
+                                 int iterations)
+{
+  int actual_count = 0;
+
+  opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+  using M = std::map<std::string, int>;
+  M m1    = {{}};
+  opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+  srand(time(0));
```

Comment:
I think we can leave this out?

Reply:
I had that in place for testing, but I suppose moving forward it isn't necessarily required. I've went ahead and removed it.